metadata-models/src/main/pegasus/com/linkedin/dataset/Constraint.pdl

```diff
@@ -0,0 +1,34 @@
+namespace com.linkedin.dataset
+
+import com.linkedin.common.Urn
+
+/**
+ * Indicates a constraint that is applied to a dataset
+ */
+record Constraint {
+
+  /**
+  * The system in which the constraints are specified
+  */
+  // e.g. "greatexpectations"
+  constraintProvider: optional string
+
+  // can be a combination of fixed values defined by Datahub, e.g. "column_value_is_not_null" etc.
+  // List of standard values TODO.  
+  // can also be a custom-format where the provider name is prefixed. e.g. "greatexpectations.fooConstraint"
+
+  constraintType: string
```

Comment:
"type"

Reply:
Still unclear what the value will contain - We should not have great expectations specific things leaking into the DataHub model. Instead we need some logical classifications on the DataHub side. For example: Is this a "table" constraint or a "column" constraint? I don't see that anywhere

Reply:
I think that's what the "constraintDomain" flag is supposed to represent